R/utils.R

```diff
@@ -1,81 +1,144 @@
+      if (file.exists(file_path)) {
+        return(file_path)
+      } else {
+        stop(dir_path, " does not contain: ", file_name)
       }
     }
-
-    if (is.null(file_name_to_load)) {
-      stop(paste("create_data_list(): No RDS or SAS files found for", file_path, x))
-    }
-
-    output <- read_file(file_path, file_name_to_load)
-
-    return(output)
   })
 
-  names(data_list) <- file_names
-
-  return(data_list)
+  # Normalize all file paths
+  return(normalizePath(unlist(file_paths)))
 }
 
 
-#' Reads RDS/SAS file and metadatas from first 6 items from file.info() its file path
-#' @param file_path a path to a file
-#' @param file_name name of a file
-#' @return a data object with an extra attribute of metadata
-read_file <- function(file_path, file_name) {
-  ext <- tools::file_ext(file_name)
 
-  if (!(toupper(ext) %in% c("RDS", "SAS7BDAT"))) {
-    stop("Usage error: read_file: file_name: file must either be RDS or SAS7BDAT.")
-  }
+#' Load Data Files
+#'
+#' This function reads data from multiple file paths and returns a list of data frames.
+#' It supports reading RDS and SAS7BDAT files.
+#'
+#' @param file_paths [character(1+)] A vector of file paths to read.
+#'
+#' @return [list] A named list of data frames, where each name is the basename of the corresponding file path.
+#'
+#' @examples
+#' path <- system.file("examples", "iris.sas7bdat", package = "haven")
+#' data_list <- load_data_files(file_paths = path)
+#' str(data_list)
+#'
+#' @export
+load_data_files <- function(file_paths) {
+  # Validate input parameters
+  checkmate::assert_character(file_paths, min.len = 1)
+  checkmate::assert_file_exists(file_paths)
+
+  # Read each file and store in a list
+  data_list <- lapply(file_paths, function(file_path) {
+    # Get file extension
+    extension <- tools::file_ext(file_path)
+
+    # Read file based on its extension
+    if (tolower(extension) == "rds") {
+      data <- readRDS(file_path)
+    } else if (tolower(extension) == "sas7bdat") {
+      data <- haven::read_sas(file_path)
+    } else {
+      stop("Unsupported file extension: ", extension)
+    }
 
-  is_rds <- toupper(ext) == "RDS"
+    # Get file metadata
+    meta <- file.info(file_path, extra_cols = FALSE)
+    meta[["path"]] <- file_path
+    meta[["file_name"]] <- basename(file_path)
 
-  file <- file.path(file_path, file_name)
-  file_name <- tools::file_path_sans_ext(file_name)
+    # Add metadata as an attribute to the data
+    rownames(data) <- NULL
+    attr(data, "meta") <- meta
 
-  # grab file info
-  meta <- file.info(file)[1L:6L]
-  meta[["path"]] <- row.names(meta)
-  meta[["file_name"]] <- file_name
-  meta <- data.frame(meta, stringsAsFactors = FALSE)
-  row.names(meta) <- NULL
+    return(data)
+  })
 
-  if (is_rds) {
-    out <- readRDS(file)
-  } else {
-    out <- haven::read_sas(file)
-  }
-  attr(out, "meta") <- meta
+  # Set names of the list elements to the basenames of the file paths
+  names(data_list) <- basename(file_paths)
```

Comment:
Whole paths would be better, because `file_paths` can point to different folders with files that share the same name.

Reply:
The use of base file names as list names is maintained to ensure backward compatibility with existing code that relies on the legacy `load_data()` function. For users who need the full file paths, this information is stored in the metadata attributes of each data frame in the returned list.

Reply:
Users of legacy `load_data()` retain the old behavior because of this statement that happens at the end of that function: 
```
names(data_list) <- file_names
```

Users of `load_data_files()` can benefit from seeing the exact path they provided as names of the output list.

Reply:
See discussion in internal chat about possibility of **removing both the paths and the extensions as well as preventing repeat entries in the resulting list**. The 
```
names(data_list) <- file_names
```
at the end of `load_data()` should make that function immune to implementing this change.